### PR TITLE
Base Object: Added method 'has_own_method()' 

### PR DESCRIPTION
**Pull request by udidol**
To determine whether a given method was declared in the current instance or was inherited by a parent class. Included unit tests.

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [X] Other... Please describe: Maintenance

## Quality assurance

- [X] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/base/base-object.php [hunk 1]
```diff
@@ -132,4 +134,28 @@ private function ensure_settings() {
...
+	 * @return bool True if the method was declared by the current instance, False if it was inherited.
+	 */
+	public function has_own_method( $method_name ) {
+		$reflection_method = new ReflectionMethod( $this, $method_name );
+
+		// If a ReflectionMethod is successfully created, get its declaring class.
+		$declaring_class = $reflection_method->getDeclaringClass();
+
+		// If the prefixed method is declared in the invoking class (and not in a parent) - it should be run.
+		// Otherwise, the new (replacement) method should be run.
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/base/base-object.php]: Removed.

File: tests/phpunit/elementor/core/base/test-base-object.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Elementor\Testing\Core\Base;
+
+use Elementor\Core\Base\Base_Object;
+use Elementor\Testing\Elementor_Test_Base;
+
+class Test_Base_Object extends Elementor_Test_Base {
+
+	public function test_has_own_method() {
+		$test_class = new Has_Own_Method_Test_Class();
```
[the remark this answers is not in the document]
Comment (udidol) [reply tests/phpunit/elementor/core/base/test-base-object.php]: Done.

File: tests/phpunit/elementor/core/base/test-base-object.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	}
+}
+
+class Has_Own_Method_Parent_Test_Class extends Base_Object {
+	public function inherited_method() {}
+
+	public function uninherited_method() {}
+}
+
+class Has_Own_Method_Test_Class extends Has_Own_Method_Parent_Test_Class {
```
[the remark this answers is not in the document]
Comment (udidol) [reply tests/phpunit/elementor/core/base/test-base-object.php]: Done.

File: core/base/base-object.php [hunk 4]
```diff
@@ -132,4 +134,28 @@ private function ensure_settings() {
 			$this->settings = $this->get_init_settings();
 		}
 	}
+
+	/**
+	 * Has Own Method
+	 *
+	 * Used for check whether the method passed as a parameter was declared in the current instance or inherited.
+	 * If it was declared by the current instance, return true. If it was inherited, call the new unprefixed method.
```
[the remark this answers is not in the document]
Comment (udidol) [reply core/base/base-object.php]: Removed.

**Review by kobizz — DISMISSED** [review]

File: core/base/base-object.php [hunk 5]
```diff
@@ -132,4 +134,25 @@ private function ensure_settings() {
...
+	 *
+	 * @since 3.1.0
+	 *
+	 * @param string $method_name
+	 *
+	 * @throws \ReflectionException
+	 * @return bool True if the method was declared by the current instance, False if it was inherited.
+	 */
+	public function has_own_method( $method_name ) {
+		$reflection_method = new ReflectionMethod( $this, $method_name );
```
Comment (matipojo) [inline core/base/base-object.php:151]: Remove the `use` from line 5.
```suggestion
		$reflection_method = new \ReflectionMethod( $this, $method_name );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/base/base-object.php (hunk 5 above)
Comment (udidol) [reply core/base/base-object.php]: Done

**Review by kobizz — APPROVED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**